cmd/kube-proxy/app/server.go

```diff
@@ -380,12 +397,34 @@ func (o *Options) loadConfigFromFile(file string) (*kubeproxyconfig.KubeProxyCon
 	return o.loadConfig(data)
 }
 
-// loadConfig decodes data as a KubeProxyConfiguration object.
+// loadConfig decodes a serialized KubeProxyConfiguration to the internal type.
 func (o *Options) loadConfig(data []byte) (*kubeproxyconfig.KubeProxyConfiguration, error) {
+
+	var (
+		configObj runtime.Object
+		gvk       *schema.GroupVersionKind
+	)
+
 	configObj, gvk, err := proxyconfigscheme.Codecs.UniversalDecoder().Decode(data, nil, nil)
 	if err != nil {
-		return nil, err
+		// Try strict decoding first. If that fails decode with a lenient
+		// decoder, which has only v1alpha1 registered, and log a warning.
+		// The lenient path is to be dropped when support for v1alpha1 is dropped.
+		if !runtime.IsStrictDecodingError(err) {
+			return nil, fmt.Errorf("failed to decode: %v", err)
+		}
+		klog.Warningf("using lenient decoding as strict decoding failed: %v", err)
```

Comment:
@liggitt had a comment (https://github.com/kubernetes/kubernetes/pull/83204/files#r337558047) about only printing this warning when we successfully decode with the lenient codec

Reply:
Updated to match